src/components/CmsZone.vue

```diff
@@ -93,7 +96,51 @@ import CmsContent from './CmsContent';
 import CmsInspectSheet from './CmsInspectSheet.vue';
 
 const durationVisibleToBeTrackedMs = 1000;
-const percentVisible = 50;
+const percentVisible = 0.5;
+
+const zones: { [key: string]: () => Promise<void> } = {};
```

Comment:
I think it'd be nice to move this and the intersection observers to a different file. I found it a little confusing that this logic is a singleton/global and the rest of the file is specific to an instance of a zone.